fix: network scene manager UNITY_INCLUDE_TESTS

NoelStephensUnity opened the pull request:
Replaces all occurrences of #if UNITY_EDITOR || DEVELOPMENT_BUILD with #if UNITY_INCLUDE_TESTS within NetworkSceneManager.cs
This fixes an issue when running as a development build and was discovered during a multiprocess test run.
This is an update to fix this issue introduced in PR-1237 that was pertaining to:
[MTT-1320](https://jira.unity3d.com/browse/MTT-1320)

NoelStephensUnity committed:
fix

Converting all references of #if UNITY_EDITOR || DEVELOPMENT_BUILD over to #if UNITY_INCLUDE_TESTS

NoelStephensUnity committed:
style

removed LF.

0xFA11 commented on com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs:
```diff
@@ -99,7 +99,7 @@ public class NetworkSceneManager : IDisposable
         // TODO: Remove `m_IsRunningUnitTest` entirely after we switch to multi-process testing
         // In MultiInstance tests, we cannot allow clients to load additional scenes as they're sharing the same scene space / Unity instance.
-#if UNITY_EDITOR || DEVELOPMENT_BUILD
+#if UNITY_INCLUDE_TESTS
```
Just to be crystal clear, did you test these changes against MultiProcess stuff? :P

NoelStephensUnity replied:
I will do that post-haste, however I know that this needed to be consistent throughout NetworkSceneManager and it was not.

NoelStephensUnity replied:
Just tested it and it both builds and works.
Compared it to develop branch and it doesn't even build.
Definitely fixes the issue.

NoelStephensUnity committed:
Merge branch 'develop' into fix/NetworkSceneManager-UNITY_INCLUDE_TESTS

NoelStephensUnity committed:
fix

Remove a check that no longer is pertinent and had a terrible comment that didn't explain what the check was for or why we didn't create and register the DDOL scene like we normally would.

NoelStephensUnity commented on com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs:
```diff
@@ -320,22 +320,10 @@ internal NetworkSceneManager(NetworkManager networkManager)
+            else
             {
-
-#if UNITY_EDITOR || DEVELOPMENT_BUILD
```
This was determined to no longer be needed, the DontDestroyOnLoadTests.ValidateNetworkObjectSynchronization validates this particular are of the code.

NoelStephensUnity committed:
Merge branch 'develop' into fix/NetworkSceneManager-UNITY_INCLUDE_TESTS
